src/url-parse-query-string.js

```diff
@@ -31,25 +33,12 @@ export function parseQueryString_(queryString) {
   if (!queryString) {
     return params;
   }
-  if (startsWith(queryString, '?') || startsWith(queryString, '#')) {
-    queryString = queryString.substr(1);
-  }
-  const pairs = queryString.split('&');
-  for (let i = 0; i < pairs.length; i++) {
-    const pair = pairs[i];
-    const eqIndex = pair.indexOf('=');
-    let name;
-    let value;
-    if (eqIndex != -1) {
-      name = decodeURIComponent(pair.substring(0, eqIndex)).trim();
-      value = decodeURIComponent(pair.substring(eqIndex + 1)).trim();
-    } else {
-      name = decodeURIComponent(pair).trim();
-      value = '';
-    }
-    if (name) {
-      params[name] = value;
-    }
+
+  let match;
+  while ((match = regex.exec(queryString))) {
+    const name = decodeURIComponent(match[1]).trim();
+    const value = match[2] ? decodeURIComponent(match[2]).trim() : '';
```

Comment:
Yeah. I'm leaning toward that as well. Not trimming is better than trimming in this case.